source/02_algo/multi_agent_cooperation_rl_eng.rst

```diff
@@ -0,0 +1,100 @@
+
+  4. Scalability: Large-scale multi-agent systems may involve high-dimensional state space and action space, which poses certain challenges for model expression ability, algorithm learning ability and hardware computing power in real scenarios.
+
+
+
+Research Direction
+-----------------------
+For the MARL cooperation task, the simplest idea is to directly apply the single-agent reinforcement learning method to the multi-agent system, that is, each agent regards other agents as factors in the environment, and still learns according to the single agent way to update the policy by interacting with the environment; This is the idea of independent Q-learning, independent PPO methods, but due to the non-stationarity of the environment and the locality of agent observations, these methods are difficult to achieve good results.
+
+At present, MARL cooperation mainly adopts the method of CTDE (centralized training and decentralized execute), and there are two main types of solutions, Valued-based MARL and Actor-Critic MARL. For details, please refer to the following figure:
+
+.. image:: images/MARL_cooperation_algo.png
+   :align: center
+   :scale: 50 %
+
+**Valued-based MARL**
+
+For Valued-based MARL, the main idea is to decompose the global reward value into local reward values that can be learned by each agent, so as to facilitate the training of the agent. There are mainly QMIX, WQMIX, QTRAN and other methods:
+
+- QMIX: The core of QMIX is to learn a monotonic Q-value mixed network, and the Q-value of each agent is summed by nonlinear transformation to generate :math:`Q_tot`. For details, please refer to `QMIX <https://github.com/opendilab/DI-engine-docs/blob/main/source/hands_on/qmix.rst>`_ [2]_
+
+- WQMIX: The core of WQMIX is the same as that of QMIX, and it also learns a Q-value mixed network, but it learns a Q-value mixed network that can break through the monotonicity limit through the weighted projection method. For details, please refer to `WQMIX <https://github.com/opendilab/DI-engine-docs/blob/main/source/hands_on/wqmix.rst>`_ [1]_
+
+- QTRAN: QTRAN breaks through the monotonicity limitation by learning independent action-value networks, hybrid action-value networks, and global state-value networks. For details, please refer to `QTRAN <https://github.com/opendilab/DI-engine-docs/blob/main/source/hands_on/qtran.rst>`_ [4]_
+
+- QPLEX: QPLEX decomposes the joint Q-value :math:`Q_tot` and the Q-value of each agent :math:`Q_i` using a Dueling structure, respectively. The IGM consistency is transformed into an easy-to-implement advantage function value range constraint, which facilitates the learning of value functions with a linear decomposition structure. For details, please refer to `QPLEX <https://arxiv.org/abs/2008.01062>`_ [10]_
+
+**Actor-critic MARL**
+
+For Actor-critic MARL, the main idea is to learn a multi-agent policy network. There are mainly COMA, MAPPO and other methods:
```

Comment:
For Actor-Critic MARL, the main idea is to use the policy gradient theorem to update the policy networks while learning a fully centralized state-action value function and use it to guide the optimization of decentralized policies.